### PR TITLE
Ucifi changes

### PR DESCRIPTION
**Pull request by paulfdietrich**
Some changes to mynewt core to prepare for Ucifi open source stack.

### CODE REVIEW

**Comment by ccollins476ad** [comment]
I have the following comments:

*lwIP static routing addon:* As I mentioned, this library was rejected when I attempted to merge it into mynewt-core earlier.  The suggestion was write our own add on.  

*tu_time_advance()*: This commit contains a hack to make things work (`extern struct os_callout_list g_callout_list;` in a source file).  I would want to do this properly before merging it.

Aside from the above, it looks good to me.  Do you want to split this PR up so that parts can be merged?

File: sys/console/full/src/uart_console.c [hunk 1]
```diff
@@ -68,6 +68,7 @@ uart_console_ring_pull_char(struct console_ring *cr)
     uint8_t ch;
 
     ch = cr->buf[cr->tail];
+    cr->buf[cr->tail] = 0xee;
```
Comment (ccollins476ad) [inline sys/console/full/src/uart_console.c]: Does `0xee` get written here just as a debugging aid?

Comment (paulfdietrich) [reply sys/console/full/src/uart_console.c]: oops. that was for debug. I can remove this.

**Comment by ccollins476ad** [comment]
I don't believe Mynewt-ucifi work is ongoing.  Please reopen if necessary.

**Pull request closed without merging**